src/main/java/uk/gov/hmcts/reform/migration/service/ChildrenDataMigrationService.java

```diff
@@ -0,0 +1,148 @@
+    public Predicate<CaseDetails> accepts() {
+        return caseDetails -> caseDetails != null && caseDetails.getData() != null &&
+            !isEmpty(caseDetails.getData().get("children"));
+    }
+
+    @Override
+    public CaseData migrate(Map<String, Object> data) {
+        CaseData caseData = objectMapper.convertValue(data, CaseData.class);
+
+        CaseData migratedCaseData = CaseData.builder()
+            .children1(migrateChildren(caseData.getChildren()))
+            .build();
+
+        log.info("new case details: {}", migratedCaseData);
+
+        return migratedCaseData;
+    }
+
+    private List<CollectionEntry<Child>> migrateChildren(OldChildren children) {
+        log.info("beginning to migrate children {}", children);
+
+        List<CollectionEntry<Child>> migratedChildren = children.getAll().stream()
+            .map(entry -> migrateIndividualChild(entry.getValue()))
+            .map(entry -> CollectionEntry.<Child>builder()
+                .id(UUID.randomUUID().toString())
+                .value(entry)
+                .build())
+            .collect(toList());
+
+        log.info("returning new structure {}", migratedChildren);
+
+        return migratedChildren;
+    }
+
+    private Child migrateIndividualChild(OldChild oc) {
+        log.info("migrating child {}", oc);
+
+        Address.AddressBuilder addressBuilder = Address.builder();
+        if (!isEmpty(oc.getAddress())) {
+            addressBuilder.addressLine1(defaultIfBlank(oc.getAddress().getAddressLine1(), null));
+            addressBuilder.addressLine2(defaultIfBlank(oc.getAddress().getAddressLine2(), null));
+            addressBuilder.addressLine3(defaultIfBlank(oc.getAddress().getAddressLine3(), null));
+            addressBuilder.postTown(defaultIfBlank(oc.getAddress().getPostTown(), null));
+            addressBuilder.postcode(defaultIfBlank(oc.getAddress().getPostcode(), null));
+            addressBuilder.county(defaultIfBlank(oc.getAddress().getCounty(), null));
+            addressBuilder.country(defaultIfBlank(oc.getAddress().getCountry(), null));
+        }
+
+        Address address = addressBuilder.build();
+
+        TelephoneNumber solicitorTelephone;
+
+        if (isEmpty(oc.getSocialWorkerTel())) {
+            solicitorTelephone = null;
+        } else {
+            TelephoneNumber.TelephoneNumberBuilder telephoneNumberBuilder = TelephoneNumber.builder();
+            telephoneNumberBuilder.telephoneNumber(defaultIfBlank(oc.getSocialWorkerTel(), null));
+            solicitorTelephone = telephoneNumberBuilder.build();
+        }
+
+        ChildParty.ChildPartyBuilder partyBuilder = ChildParty.builder();
+        partyBuilder.partyId(UUID.randomUUID().toString());
+
+        if (!isEmpty(oc.getChildName())) {
+            partyBuilder.firstName(splitName(oc.getChildName()).get(0));
+
+            if (splitName(oc.getChildName()).size() > 1) {
+                partyBuilder.lastName(splitName(oc.getChildName()).get(1));
```

Comment:
Any reason why this we are not using defaultIfBlank null in the firstName lastName scenario?

String " Adam" "Adam " will lead either firstName or LastName being blank in those two scenarios - not sure if a problem, but it is inconsistency.

Reply:
I am confident this is fine for throw away code. It will add all names bar the last to first name and then the last string to last name